composition-go/composition.go

```diff
@@ -41,9 +50,60 @@ var (
 	pool sync.Pool
 )
 
+func updateSchemas(subgraphs []*Subgraph) ([]*Subgraph, error) {
+	updatedSubgraphs := make([]*Subgraph, 0, len(subgraphs))
+	for _, subgraph := range subgraphs {
+		if subgraph.Schema != "" {
+			updatedSubgraphs = append(updatedSubgraphs, subgraph)
+			continue
+		}
+		req, err := http.NewRequest("POST", subgraph.URL, strings.NewReader(sdlQuery))
+		if err != nil {
+			return nil, fmt.Errorf("could not create request retrieving schema for subgraph %s: %w", subgraph.Name, err)
+		}
+		req.Header.Add("Content-Type", "application/json")
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve schema for subgraph %s: %w", subgraph.Name, err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("could not retrieve schema for subgraph %s: unexpected status code %d", subgraph.Name, resp.StatusCode)
+		}
+		var sdlResult struct {
```

Comment:
How about extracting this into a struct?

Reply:
Sounds reasonable, I've moved it to a type